spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -557,6 +561,34 @@ else if (listener instanceof MessageListener) {
 					KafkaMessageListenerContainer.this.consumerFactory.getConfigurationProperties();
 			this.checkNullKeyForExceptions = checkDeserializer(findDeserializerClass(props, false));
 			this.checkNullValueForExceptions = checkDeserializer(findDeserializerClass(props, true));
+			this.syncCommitTimeout = determineSyncCommitTimeout();
+			if (this.containerProperties.getSyncCommitTimeout() == null) {
+				this.containerProperties.setSyncCommitTimeout(this.syncCommitTimeout);
+			}
+		}
+
+		private Duration determineSyncCommitTimeout() {
+			if (this.containerProperties.getSyncCommitTimeout() != null) {
+				return this.containerProperties.getSyncCommitTimeout();
+			}
+			else {
+				Object timeout = this.containerProperties.getConsumerProperties()
+						.get(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
+				if (timeout == null) {
+					timeout = KafkaMessageListenerContainer.this.consumerFactory.getConfigurationProperties()
+							.get(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
+				}
+				if (timeout instanceof Number) {
+					return Duration.ofMillis(((Number) timeout).longValue());
+				}
+				else if (timeout instanceof String) {
+					return Duration.ofMillis(Long.parseLong((String) timeout));
+				}
+				else {
+					return Duration.ofSeconds(60);
```

Comment:
Can the `timeout` from properties be as a `Duration` as well? 
Why there is no such an `else if` then?
Why do you default it to `60 secs` and no any WARN that  `timeout` is not of expected type?

Reply:
Good point; I default it to 60 seconds because that's what the client defaults it to if it is not provided. But, I agree, it should only be that if it's `null`.

Reply:
Actually, I don't think Kafka will accept `Duration` there, but I'll add the code anyway.